integration/spark2/src/main/scala/org/apache/carbondata/store/SparkCarbonStore.scala

```diff
@@ -110,8 +110,16 @@ class SparkCarbonStore extends MetaCachedCarbonStore {
   }
 
   def startSearchMode(): Unit = {
+    LOG.info("Starting search mode master")
     master = new Master(session.sparkContext.getConf)
     master.startService()
+    var count = 0
+    while (master.isStarted().compareAndSet(false, false) && count < 5000) {
```

Comment:
I think you can move this while loop into master.startService, so that master.startService is a blocking operation until it is success.  No need to expose master.isStarted method

Reply:
ok，I will try